Thunk assertion fail on Go sample (ELF 64-bit LSB shared object, ARM aarch64)

For https://www.virustotal.com/gui/file/50f107898c6472a1155354f74b039ce951243cdb9714ef881a917d69b8a71ee0/ and associated BinExport file (available on request).

```
Traceback (most recent call last):
  File "/usr/local/google/home/moritzraabe/code/capa/.venv/bin/capa", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/google/home/moritzraabe/code/capa/capa/main.py", line 990, in main
    extractor = get_extractor_from_cli(args, input_format, backend)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/google/home/moritzraabe/code/capa/capa/main.py", line 837, in get_extractor_from_cli
    extractor = capa.loader.get_extractor(
                ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/google/home/moritzraabe/code/capa/capa/loader.py", line 316, in get_extractor
    return capa.features.extractors.binexport2.extractor.BinExport2FeatureExtractor(be2, buf)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/google/home/moritzraabe/code/capa/capa/features/extractors/binexport2/extractor.py", line 47, in __init__
    self.analysis: BinExport2Analysis = BinExport2Analysis(self.be2, self.idx, self.buf)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/google/home/moritzraabe/code/capa/capa/features/extractors/binexport2/__init__.py", line 259, in __init__
    self._compute_thunks()
  File "/usr/local/google/home/moritzraabe/code/capa/capa/features/extractors/binexport2/__init__.py", line 287, in _compute_thunks
    assert len(thunk_callees) == 1, f"thunk @ {hex(addr)} failed"
           ^^^^^^^^^^^^^^^^^^^^^^^
AssertionError: thunk @ 0x42d8c0 failed

```

From what I see right now, there's a thunk function identified at VA 0x42d8c0 but this appears to be a regular function so our assertion fails.

Easiest fix: remove the assertion, then the file processes.

I'm afraid a deeper analysis could take some time and could require updates to Ghidra and/or BinExport.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -45,6 +45,7 @@
 - cape: make Process model flexible and procmemory optional to load newest reports #2466 @mr-tz
 - binja: fix unit test failure by fixing up the analysis for file al-khaser_x64.exe_ #2507 @xusheng6
 - binja: move the stack string detection to function level #2516 @xusheng6
+- BinExport2: fix handling of incorrect thunk functions #2524 @williballenthin
 
 ### capa Explorer Web
 

--- a/capa/features/extractors/binexport2/__init__.py
+++ b/capa/features/extractors/binexport2/__init__.py
@@ -280,11 +280,13 @@ class BinExport2Analysis:
             curr_idx: int = idx
             for _ in range(capa.features.common.THUNK_CHAIN_DEPTH_DELTA):
                 thunk_callees: list[int] = self.idx.callees_by_vertex_index[curr_idx]
-                # if this doesn't hold, then it doesn't seem like this is a thunk,
+                # If this doesn't hold, then it doesn't seem like this is a thunk,
                 # because either, len is:
-                #    0 and the thunk doesn't point to anything, or
+                #    0 and the thunk doesn't point to anything or is indirect, like `call eax`, or
                 #   >1 and the thunk may end up at many functions.
-                assert len(thunk_callees) == 1, f"thunk @ {hex(addr)} failed"
+                # In any case, this doesn't appear to be the sort of thunk we're looking for.
+                if len(thunk_callees) != 1:
+                    break
 
                 thunked_idx: int = thunk_callees[0]
                 thunked_vertex: BinExport2.CallGraph.Vertex = self.be2.call_graph.vertex[thunked_idx]

